src/lib/pio.pl

```diff
@@ -163,13 +169,16 @@
     ;   partial_string_last_tail(PartialStringTail0, PartialStringTail)
     ).
 
-string_get_n_chars([], _, _, []).
-string_get_n_chars([S|Ss], BufferPos, N, Chars) :-
-    (   BufferPos = 0 ->
-        string_get_n_chars_([S|Ss], N, Chars)
-    ;   BufferPos1 is BufferPos - 1,
-        string_get_n_chars(Ss, BufferPos1, N, Chars)
-    ).
+string_get_n_chars(String, Pos, N, Chars) :-
+    chars_to_read(CharsToRead),
+    (   CharsToRead < Pos -> 
+        % I have absolutely no idea why this is needed (maybe it's a bug?),
```

Comment:
Could you please expand on this: What is unexpectedly needed here? There may be indeed a mistake in the implementation of `'$skip_max_list'/4`: If something does not work as expected, please consider filing it as a separate issue with self-contained example.

Reply:
Filed in #2037.

Reply:
Thank you a lot! It turned out to be an actual implementation mistake in `'$skip_max_list'/4` which is now already corrected! Please remain vigilant: In the areas you are working on, you will be able to find implementation mistakes that nobody else will ever see, please file everything that does not work as expected as an issue!

Reply:
I did not file it immediately because I couldn't reproduce it in a small contained example. Turns out I only tested for a sequence of 2 partial strings, but the problem only manifested itself with 3 or more. Sometimes bugs are really elusive.